### PR TITLE
Function Calling

### PR DESCRIPTION
**Pull request by nsosio**

### CODE REVIEW

**Review by pritamsoni-hsr — APPROVED** [review]
LGTM, just few suggestions

File: prem_utils/connectors/anthropic.py [hunk 1]
```diff
@@ -82,6 +83,40 @@ def preprocess_messages(self, messages):
...
+
+    def _parse_tools(self, tools: list[dict[str, any]]):
+        if not tools:
+            return []
+        parsed_tools = []
+
+        for tool in tools:
+            transformed_tool = {
+                "name": tool["function"]["name"],
+                "description": tool["function"]["description"],
```
[the remark this answers is not in the document]
Comment (nsosio) [reply prem_utils/connectors/anthropic.py]: done

File: prem_utils/connectors/anthropic.py [hunk 2]
```diff
@@ -82,6 +83,40 @@ def preprocess_messages(self, messages):
                 filtered_messages.append(message)
         return system_prompt, filtered_messages
 
+    def _get_content(self, response, tools=False):
+        if tools:
+            return {"content": response.content[0].text, "role": "assistant", "tool_calls": []}
+        else:
+            tool_messages = filter(lambda x: "input" in x.__dir__() and "name" in x.__dir__(), response.content)
```
[the remark this answers is not in the document]
Comment (nsosio) [reply prem_utils/connectors/anthropic.py]: done, thanks 👍

**Review by pritamsoni-hsr — APPROVED** [review]
LGTM

**Pull request merged**